### PR TITLE
FileSelNative::Execute() fix: default filename was some trash on saving

### PR DESCRIPTION
**Pull request by zsdebre**
[BUG + FIX: FileSelNative shows default filename as some trash characters](https://www.ultimatepp.org/forums/index.php?t=msg&th=11901&goto=58798&#msg_58798) [[message #58798](https://www.ultimatepp.org/forums/index.php?t=msg&th=11901&goto=58798&#msg_58798)]

### CODE REVIEW

File: uppsrc/CtrlLib/Win32.cpp [hunk 1]
```diff
@@ -206,14 +206,14 @@ bool FileSelNative::Execute(bool open, const char *dlgtitle) {
...
-						out << W32(fn);
-					out.Cat(0);
+						out.Append(ToSystemCharsetW(fn));
+					out.Add(0);
 				}
 			}
 		}
-		int l = min(out.GetLength(), bufsize - 1);
+		int l = min(out.GetCount()*2, bufsize - 1);
 		memcpy(buffer, out, l + 1);
```
Comment (mirek-fidler) [inline uppsrc/CtrlLib/Win32.cpp]: I feel like "l + 1" here might be wrong, in fact the whole handling of terminal double \0 does not feel right...

Comment (zsdebre) [reply uppsrc/CtrlLib/Win32.cpp]: Yes, you are absolutely right, thanks for your feelings :)
The "+ 1" is not needed, nor the "out.Add(0);" line.
I fixed in the branch.

**Comment by mirek-fidler** [comment]
Uhm, those NULL characters added to the list are for multiple filenames.

https://docs.microsoft.com/en-us/windows/win32/api/commdlg/ns-commdlg-openfilenamew

You are supposed to separate files with NULL, then end the list with double NULL...

**Comment by mirek-fidler** [comment]
Fixed with slightly modified code...

**Comment by zsdebre** [comment]
Thanks!

**Pull request closed without merging**